# TiVo

### Focus
TiVo value propositon and how to measure it

### Qs
- does the study evaluate consumer attitudes well?
- what are strengths/ weaknesses of the present approach?
- would you do anything diff in the market research project? what would you have done and what would you expect to find?
- why segment in the first place? best way?
- how do diff kinds of research relate?

In [2]:
# import libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
#import seaborn as sns
import json
from pandas.io.json import json_normalize
from datetime import datetime
from sklearn import preprocessing

In [3]:
df = pd.read_csv("TiVoData.csv")

In [4]:
df.head()

ID  Gender Marital Status   Work Status Education  \
0  1.0    male        married  professional      none   
1  2.0    male         single          none      none   
2  3.0    male        married  professional        BA   
3  4.0    male        married          none       PhD   
4  5.0  female         single          none      none   

   Annual Income (x1000 $)   Age       Location Purchasing Decision-maker  \
0                     49.0  30.0        Florida                    family   
1                     46.0  36.0        Alabama                    single   
2                     58.0  66.0  Massachusetts                    family   
3                     51.0  78.0       New York                    family   
4                     46.0  52.0        Montana                    single   

         Purchasing Location  Monthly Electronics Spend  \
0  mass-consumer electronics                       35.0   
1  mass-consumer electronics                       35.0   
2           specialty stores                       64.0   
3  mass-consumer electronics                       33.0   
4  mass-consumer electronics                       45.0   

   Monthly Household Spend  Purchasing Frequency (every x months)  \
0                    150.0                                   13.0   
1                    163.0                                   26.0   
2                    103.0                                   13.0   
3                    154.0                                   22.0   
4                    161.0                                   47.0   

  Technology Adoption  TV Viewing (hours/day)  \
0                late                     2.0   
1                late                    10.0   
2               early                     0.0   
3                late                     5.0   
4                late                     2.0   

                             Favorite feature  
0  saving favorite shows to watch as a family  
1  saving favorite shows to watch as a family  
2                               time shifting  
3  saving favorite shows to watch as a family  
4  saving favorite shows to watch as a family

In [5]:
df.tail()

ID  Gender Marital Status   Work Status Education  \
998    999.0    male        married          none        BA   
999   1000.0  female        married  professional      none   
1000     NaN     NaN            NaN           NaN       NaN   
1001     NaN     NaN            NaN           NaN       NaN   
1002     NaN     NaN            NaN           NaN       NaN   

      Annual Income (x1000 $)   Age   Location Purchasing Decision-maker  \
998                      49.0  67.0      Maine                    family   
999                      33.0  58.0  Tennessee                    single   
1000                      NaN   NaN        NaN                       NaN   
1001                     21.0  18.0        NaN                       NaN   
1002                    730.0  80.0        NaN                       NaN   

            Purchasing Location  Monthly Electronics Spend  \
998   mass-consumer electronics                       41.0   
999                      retail                       23.0   
1000                        NaN                        NaN   
1001                        NaN                        NaN   
1002                        NaN                        NaN   

      Monthly Household Spend  Purchasing Frequency (every x months)  \
998                      55.0                                   23.0   
999                     104.0                                    1.0   
1000                      NaN                                    NaN   
1001                      NaN                                    NaN   
1002                      NaN                                    NaN   

     Technology Adoption  TV Viewing (hours/day)  \
998                 late                    10.0   
999                early                     1.0   
1000                 NaN                     NaN   
1001                 NaN                     NaN   
1002                 NaN                     NaN   

                                Favorite feature  
998   saving favorite shows to watch as a family  
999                                  cool gadget  
1000                                         NaN  
1001                                         NaN  
1002                                         NaN

In [6]:
df['Annual Spending on Electronics'] = df['Monthly Electronics Spend'] * 12
df.head()

ID  Gender Marital Status   Work Status Education  \
0  1.0    male        married  professional      none   
1  2.0    male         single          none      none   
2  3.0    male        married  professional        BA   
3  4.0    male        married          none       PhD   
4  5.0  female         single          none      none   

   Annual Income (x1000 $)   Age       Location Purchasing Decision-maker  \
0                     49.0  30.0        Florida                    family   
1                     46.0  36.0        Alabama                    single   
2                     58.0  66.0  Massachusetts                    family   
3                     51.0  78.0       New York                    family   
4                     46.0  52.0        Montana                    single   

         Purchasing Location  Monthly Electronics Spend  \
0  mass-consumer electronics                       35.0   
1  mass-consumer electronics                       35.0   
2           specialty stores                       64.0   
3  mass-consumer electronics                       33.0   
4  mass-consumer electronics                       45.0   

   Monthly Household Spend  Purchasing Frequency (every x months)  \
0                    150.0                                   13.0   
1                    163.0                                   26.0   
2                    103.0                                   13.0   
3                    154.0                                   22.0   
4                    161.0                                   47.0   

  Technology Adoption  TV Viewing (hours/day)  \
0                late                     2.0   
1                late                    10.0   
2               early                     0.0   
3                late                     5.0   
4                late                     2.0   

                             Favorite feature  Annual Spending on Electronics  
0  saving favorite shows to watch as a family                           420.0  
1  saving favorite shows to watch as a family                           420.0  
2                               time shifting                           768.0  
3  saving favorite shows to watch as a family                           396.0  
4  saving favorite shows to watch as a family                           540.0

In [7]:
df['Spending as % of Income'] = round(df['Annual Spending on Electronics'] / (df['Annual Income (x1000 $)']*1000),4)
df.head()

ID  Gender Marital Status   Work Status Education  \
0  1.0    male        married  professional      none   
1  2.0    male         single          none      none   
2  3.0    male        married  professional        BA   
3  4.0    male        married          none       PhD   
4  5.0  female         single          none      none   

   Annual Income (x1000 $)   Age       Location Purchasing Decision-maker  \
0                     49.0  30.0        Florida                    family   
1                     46.0  36.0        Alabama                    single   
2                     58.0  66.0  Massachusetts                    family   
3                     51.0  78.0       New York                    family   
4                     46.0  52.0        Montana                    single   

         Purchasing Location  Monthly Electronics Spend  \
0  mass-consumer electronics                       35.0   
1  mass-consumer electronics                       35.0   
2           specialty stores                       64.0   
3  mass-consumer electronics                       33.0   
4  mass-consumer electronics                       45.0   

   Monthly Household Spend  Purchasing Frequency (every x months)  \
0                    150.0                                   13.0   
1                    163.0                                   26.0   
2                    103.0                                   13.0   
3                    154.0                                   22.0   
4                    161.0                                   47.0   

  Technology Adoption  TV Viewing (hours/day)  \
0                late                     2.0   
1                late                    10.0   
2               early                     0.0   
3                late                     5.0   
4                late                     2.0   

                             Favorite feature  Annual Spending on Electronics  \
0  saving favorite shows to watch as a family                           420.0   
1  saving favorite shows to watch as a family                           420.0   
2                               time shifting                           768.0   
3  saving favorite shows to watch as a family                           396.0   
4  saving favorite shows to watch as a family                           540.0   

   Spending as % of Income  
0                   0.0086  
1                   0.0091  
2                   0.0132  
3                   0.0078  
4                   0.0117

In [8]:
print(df['Gender'].value_counts()/df['Gender'].count())

male      0.535
female    0.465
Name: Gender, dtype: float64


In [9]:
print(df['Marital Status'].value_counts()/df['Marital Status'].count())

married    0.72
single     0.28
Name: Marital Status, dtype: float64


In [10]:
print(df['Work Status'].value_counts()/df['Work Status'].count())

professional    0.65
none            0.35
Name: Work Status, dtype: float64


In [11]:
print(df['Education'].value_counts()/df['Education'].count())

none    0.506
BA      0.260
MA      0.124
PhD     0.110
Name: Education, dtype: float64


In [12]:
ranges = [0,40,60,80,800]
print(df['Annual Income (x1000 $)'].groupby(pd.cut(df['Annual Income (x1000 $)'],ranges)).count())

Annual Income (x1000 $)
(0, 40]      632
(40, 60]     358
(60, 80]       9
(80, 800]      3
Name: Annual Income (x1000 $), dtype: int64


In [13]:
ranges = [0,20,40,60,80]
print(df['Annual Income (x1000 $)'].groupby(pd.cut(df['Annual Income (x1000 $)'],ranges)).count())

Annual Income (x1000 $)
(0, 20]       0
(20, 40]    632
(40, 60]    358
(60, 80]      9
Name: Annual Income (x1000 $), dtype: int64


In [14]:
print(df['Location'].value_counts()/df['Location'].count())

Pennsylvania      0.050
Hawaii            0.045
Delaware          0.045
Arizona           0.043
Washington        0.043
Idaho             0.043
Vermont           0.043
California        0.041
Maine             0.041
Colorado          0.039
New York          0.039
Maryland          0.039
Massachusetts     0.038
Nevada            0.037
Connecticut       0.037
Florida           0.035
New Jersey        0.035
New Hampshire     0.035
Illinois          0.034
Tennessee         0.032
Georgia           0.027
Rhode Island      0.027
Oregon            0.010
North Dakota      0.010
Utah              0.009
Missouri          0.008
North Carolina    0.008
West Virginia     0.008
Texas             0.007
Montana           0.007
South Carolina    0.006
Indiana           0.006
Michigan          0.006
Kansas            0.006
Kentucky          0.006
Alabama           0.006
Louisiana         0.005
Alaska            0.005
New Mexico        0.005
Wyoming           0.004
South Dakota      0.004
Wisconsin       

In [15]:
print(df['Purchasing Decision-maker'].value_counts()/df['Purchasing Decision-maker'].count())

family    0.56
single    0.44
Name: Purchasing Decision-maker, dtype: float64


In [16]:
print(df['Purchasing Location'].value_counts()/df['Purchasing Location'].count())

retail                       0.294
discount                     0.293
mass-consumer electronics    0.200
specialty stores             0.170
web (ebay)                   0.043
Name: Purchasing Location, dtype: float64


In [17]:
ranges = [0,40,60,80,100]
print(df['Monthly Electronics Spend'].groupby(pd.cut(df['Monthly Electronics Spend'],ranges)).count())

Monthly Electronics Spend
(0, 40]      733
(40, 60]     196
(60, 80]      62
(80, 100]      9
Name: Monthly Electronics Spend, dtype: int64


In [18]:
ranges = [0,50,100,150,200,250,300,350,400]
print(df['Monthly Household Spend'].groupby(pd.cut(df['Monthly Household Spend'],ranges)).count())

Monthly Household Spend
(0, 50]       281
(50, 100]     389
(100, 150]    156
(150, 200]     96
(200, 250]     42
(250, 300]     16
(300, 350]     11
(350, 400]      9
Name: Monthly Household Spend, dtype: int64


In [19]:
ranges = [0,10,20,30,40,50]
print(df['Purchasing Frequency (every x months)'].groupby(pd.cut(df['Purchasing Frequency (every x months)'],ranges)).count())

Purchasing Frequency (every x months)
(0, 10]     241
(10, 20]    231
(20, 30]    187
(30, 40]    192
(40, 50]    149
Name: Purchasing Frequency (every x months), dtype: int64


In [20]:
print(df['Technology Adoption'].value_counts()/df['Technology Adoption'].count())

early    0.8
late     0.2
Name: Technology Adoption, dtype: float64


In [21]:
ranges = [0,5,10,15]
print(df['TV Viewing (hours/day)'].groupby(pd.cut(df['TV Viewing (hours/day)'],ranges)).count())

TV Viewing (hours/day)
(0, 5]      715
(5, 10]      65
(10, 15]     45
Name: TV Viewing (hours/day), dtype: int64


In [22]:
print(df['Favorite feature'].value_counts()/df['Favorite feature'].count())

cool gadget                                   0.228
time shifting                                 0.221
schedule control                              0.221
saving favorite shows to watch as a family    0.200
programming/interactive features              0.130
Name: Favorite feature, dtype: float64


In [23]:
ranges = [0,250,500,1000]
print(df['Annual Spending on Electronics'].groupby(pd.cut(df['Annual Spending on Electronics'],ranges)).count())

Annual Spending on Electronics
(0, 250]       444
(250, 500]     300
(500, 1000]    251
Name: Annual Spending on Electronics, dtype: int64


In [24]:
ranges = [0,.005,.010,.015,0.02]
print(df['Spending as % of Income'].groupby(pd.cut(df['Spending as % of Income'],ranges)).count())

Spending as % of Income
(0.0, 0.005]      57
(0.005, 0.01]    558
(0.01, 0.015]    317
(0.015, 0.02]     68
Name: Spending as % of Income, dtype: int64


In [25]:
df.loc[(df["Gender"] == "male") & (df["Marital Status"] == "married")]

ID Gender Marital Status   Work Status Education  \
0      1.0   male        married  professional      none   
2      3.0   male        married  professional        BA   
3      4.0   male        married          none       PhD   
6      7.0   male        married  professional      none   
7      8.0   male        married          none      none   
8      9.0   male        married  professional      none   
12    13.0   male        married  professional        BA   
16    17.0   male        married  professional        BA   
23    24.0   male        married          none      none   
27    28.0   male        married  professional      none   
34    35.0   male        married  professional      none   
36    37.0   male        married          none      none   
39    40.0   male        married          none      none   
47    48.0   male        married  professional        BA   
49    50.0   male        married          none      none   
50    51.0   male        married          none      none   
51    52.0   male        married          none      none   
52    53.0   male        married          none      none   
58    59.0   male        married  professional        BA   
59    60.0   male        married  professional        MA   
61    62.0   male        married          none       PhD   
62    63.0   male        married  professional       PhD   
67    68.0   male        married  professional       PhD   
69    70.0   male        married  professional      none   
70    71.0   male        married          none        BA   
73    74.0   male        married  professional      none   
78    79.0   male        married  professional        BA   
81    82.0   male        married          none      none   
82    83.0   male        married  professional      none   
86    87.0   male        married  professional        MA   
..     ...    ...            ...           ...       ...   
935  936.0   male        married  professional        BA   
937  938.0   male        married          none      none   
939  940.0   male        married          none      none   
942  943.0   male        married          none       PhD   
943  944.0   male        married          none      none   
946  947.0   male        married          none        BA   
947  948.0   male        married  professional      none   
948  949.0   male        married          none      none   
951  952.0   male        married  professional      none   
956  957.0   male        married          none      none   
957  958.0   male        married          none      none   
959  960.0   male        married          none      none   
961  962.0   male        married  professional      none   
962  963.0   male        married  professional        BA   
963  964.0   male        married  professional        BA   
969  970.0   male        married          none      none   
971  972.0   male        married  professional       PhD   
972  973.0   male        married          none        BA   
973  974.0   male        married          none      none   
977  978.0   male        married  professional      none   
983  984.0   male        married          none      none   
984  985.0   male        married          none      none   
986  987.0   male        married          none        BA   
988  989.0   male        married          none        MA   
990  991.0   male        married  professional      none   
993  994.0   male        married  professional      none   
994  995.0   male        married  professional        BA   
996  997.0   male        married  professional        MA   
997  998.0   male        married  professional      none   
998  999.0   male        married          none        BA   

     Annual Income (x1000 $)   Age        Location Purchasing Decision-maker  \
0                       49.0  30.0         Florida                    family   
2                       58.0  66.0   Massachusetts                    family   
3                       51.0  78.0        New Y

In [26]:
df.loc[(df["Gender"] == "male") & (df["Marital Status"] == "married") & (df["Technology Adoption"] == "early")]

ID Gender Marital Status   Work Status Education  \
2      3.0   male        married  professional        BA   
6      7.0   male        married  professional      none   
7      8.0   male        married          none      none   
8      9.0   male        married  professional      none   
12    13.0   male        married  professional        BA   
16    17.0   male        married  professional        BA   
27    28.0   male        married  professional      none   
36    37.0   male        married          none      none   
39    40.0   male        married          none      none   
47    48.0   male        married  professional        BA   
49    50.0   male        married          none      none   
50    51.0   male        married          none      none   
51    52.0   male        married          none      none   
52    53.0   male        married          none      none   
59    60.0   male        married  professional        MA   
67    68.0   male        married  professional       PhD   
69    70.0   male        married  professional      none   
70    71.0   male        married          none        BA   
73    74.0   male        married  professional      none   
78    79.0   male        married  professional        BA   
81    82.0   male        married          none      none   
82    83.0   male        married  professional      none   
87    88.0   male        married          none      none   
94    95.0   male        married  professional       PhD   
98    99.0   male        married  professional      none   
102  103.0   male        married  professional      none   
104  105.0   male        married          none      none   
107  108.0   male        married          none      none   
108  109.0   male        married  professional        BA   
110  111.0   male        married          none      none   
..     ...    ...            ...           ...       ...   
922  923.0   male        married  professional      none   
924  925.0   male        married          none      none   
927  928.0   male        married          none      none   
929  930.0   male        married  professional      none   
934  935.0   male        married  professional      none   
937  938.0   male        married          none      none   
939  940.0   male        married          none      none   
943  944.0   male        married          none      none   
946  947.0   male        married          none        BA   
947  948.0   male        married  professional      none   
948  949.0   male        married          none      none   
951  952.0   male        married  professional      none   
956  957.0   male        married          none      none   
957  958.0   male        married          none      none   
959  960.0   male        married          none      none   
961  962.0   male        married  professional      none   
962  963.0   male        married  professional        BA   
963  964.0   male        married  professional        BA   
969  970.0   male        married          none      none   
971  972.0   male        married  professional       PhD   
972  973.0   male        married          none        BA   
973  974.0   male        married          none      none   
977  978.0   male        married  professional      none   
983  984.0   male        married          none      none   
984  985.0   male        married          none      none   
990  991.0   male        married  professional      none   
993  994.0   male        married  professional      none   
994  995.0   male        married  professional        BA   
996  997.0   male        married  professional        MA   
997  998.0   male        married  professional      none   

     Annual Income (x1000 $)   Age        Location Purchasing Decision-maker  \
2                       58.0  66.0   Massachusetts                    family   
6                       33.0  62.0      California                    single   
7                       29.0  30.0   New Hampsh

In [27]:
df.loc[(df["Gender"] == "male") & (df["Marital Status"] == "married") & (df["Technology Adoption"] == "early") & (df["Monthly Electronics Spend"]*24 > 500)]

ID Gender Marital Status   Work Status Education  \
2      3.0   male        married  professional        BA   
7      8.0   male        married          none      none   
8      9.0   male        married  professional      none   
12    13.0   male        married  professional        BA   
47    48.0   male        married  professional        BA   
59    60.0   male        married  professional        MA   
69    70.0   male        married  professional      none   
73    74.0   male        married  professional      none   
78    79.0   male        married  professional        BA   
81    82.0   male        married          none      none   
94    95.0   male        married  professional       PhD   
102  103.0   male        married  professional      none   
108  109.0   male        married  professional        BA   
125  126.0   male        married  professional        BA   
133  134.0   male        married  professional        BA   
138  139.0   male        married  professional      none   
144  145.0   male        married  professional      none   
147  148.0   male        married  professional        BA   
151  152.0   male        married  professional        BA   
154  155.0   male        married  professional        BA   
158  159.0   male        married  professional        BA   
171  172.0   male        married  professional      none   
172  173.0   male        married  professional        BA   
173  174.0   male        married  professional        BA   
187  188.0   male        married  professional        BA   
188  189.0   male        married  professional        BA   
197  198.0   male        married  professional        MA   
203  204.0   male        married  professional        MA   
207  208.0   male        married  professional        MA   
208  209.0   male        married          none      none   
..     ...    ...            ...           ...       ...   
775  776.0   male        married  professional        BA   
783  784.0   male        married  professional        BA   
785  786.0   male        married  professional        MA   
788  789.0   male        married          none      none   
793  794.0   male        married  professional       PhD   
800  801.0   male        married  professional      none   
813  814.0   male        married  professional       PhD   
814  815.0   male        married  professional        BA   
816  817.0   male        married  professional       PhD   
818  819.0   male        married          none      none   
820  821.0   male        married          none      none   
826  827.0   male        married  professional        MA   
831  832.0   male        married  professional        BA   
837  838.0   male        married  professional      none   
838  839.0   male        married  professional        BA   
845  846.0   male        married  professional        MA   
848  849.0   male        married  professional        BA   
858  859.0   male        married  professional        MA   
867  868.0   male        married  professional        BA   
883  884.0   male        married  professional        BA   
888  889.0   male        married  professional        BA   
900  901.0   male        married  professional      none   
904  905.0   male        married  professional        BA   
922  923.0   male        married  professional      none   
937  938.0   male        married          none      none   
956  957.0   male        married          none      none   
962  963.0   male        married  professional        BA   
963  964.0   male        married  professional        BA   
971  972.0   male        married  professional       PhD   
994  995.0   male        married  professional        BA   

     Annual Income (x1000 $)   Age        Location Purchasing Decision-maker  \
2                       58.0  66.0   Massachusetts                    family   
7                       29.0  30.0   New Hampshire                    single   
8                       57.0  60.0   Massachuse

Married Men who are early adopters of tech that can afford TiVo (extrapolated from 2 years worth of their monthly spend): 134

In [28]:
wm = df.loc[(df["Gender"] == "female") & (df["Education"].isin(["MA","PhD"])) & ( (df["Marital Status"] == "single") | (df["Purchasing Decision-maker"] == "single"))]

In [29]:
len(wm)

56

Women that make individual purchasing decisions on tech (educated + single/ makes decisions w/o spousal input): 56

In [30]:
df.loc[(df["Technology Adoption"] == "early") & (df["Purchasing Frequency (every x months)"] <= 12) & (df["Purchasing Location"] == "specialty stores")]

ID  Gender Marital Status   Work Status Education  \
8      9.0    male        married  professional      none   
12    13.0    male        married  professional        BA   
30    31.0  female         single  professional        BA   
41    42.0  female        married  professional        BA   
47    48.0    male        married  professional        BA   
59    60.0    male        married  professional        MA   
76    77.0    male         single  professional        BA   
78    79.0    male        married  professional        BA   
83    84.0  female        married  professional        BA   
94    95.0    male        married  professional       PhD   
100  101.0    male         single  professional        BA   
102  103.0    male        married  professional      none   
108  109.0    male        married  professional        BA   
111  112.0    male         single  professional        BA   
117  118.0  female        married  professional       PhD   
133  134.0    male        married  professional        BA   
146  147.0  female        married  professional        BA   
147  148.0    male        married  professional        BA   
151  152.0    male        married  professional        BA   
154  155.0    male        married  professional        BA   
158  159.0    male        married  professional        BA   
160  161.0  female         single  professional      none   
165  166.0  female        married  professional      none   
171  172.0    male        married  professional      none   
172  173.0    male        married  professional        BA   
187  188.0    male        married  professional        BA   
188  189.0    male        married  professional        BA   
197  198.0    male        married  professional        MA   
204  205.0  female        married  professional        BA   
217  218.0  female        married  professional        BA   
..     ...     ...            ...           ...       ...   
739  740.0  female        married  professional        BA   
742  743.0    male        married  professional       PhD   
747  748.0    male        married  professional       PhD   
774  775.0    male        married  professional        BA   
775  776.0    male        married  professional        BA   
786  787.0  female        married  professional      none   
790  791.0  female         single  professional        MA   
793  794.0    male        married  professional       PhD   
800  801.0    male        married  professional      none   
807  808.0    male         single  professional        BA   
814  815.0    male        married  professional        BA   
826  827.0    male        married  professional        MA   
836  837.0    male         single  professional        BA   
838  839.0    male        married  professional        BA   
848  849.0    male        married  professional        BA   
858  859.0    male        married  professional        MA   
866  867.0    male         single  professional        BA   
867  868.0    male        married  professional        BA   
879  880.0  female        married  professional        BA   
883  884.0    male        married  professional        BA   
888  889.0    male        married  professional        BA   
898  899.0  female        married  professional        MA   
904  905.0    male        married  professional        BA   
922  923.0    male        married  professional      none   
928  929.0  female        married  professional       PhD   
958  959.0    male         single  professional      none   
962  963.0    male        married  professional        BA   
963  964.0    male        married  professional        BA   
971  972.0    male        married  professional       PhD   
994  995.0    male        married  professional        BA   

     Annual Income (x1000 $)   Age        Location Purchasing Decision-maker  \
8                       57.0  60.0   Massachusetts                    family   
12                      64.0  80.0      New Jersey               

early adopters that buy tech onve a year in specialty stores: 132

In [31]:
df.drop([1000, 1001,1002])

ID  Gender Marital Status   Work Status Education  \
0       1.0    male        married  professional      none   
1       2.0    male         single          none      none   
2       3.0    male        married  professional        BA   
3       4.0    male        married          none       PhD   
4       5.0  female         single          none      none   
5       6.0  female        married          none        BA   
6       7.0    male        married  professional      none   
7       8.0    male        married          none      none   
8       9.0    male        married  professional      none   
9      10.0  female        married  professional      none   
10     11.0  female         single  professional       PhD   
11     12.0  female        married          none      none   
12     13.0    male        married  professional        BA   
13     14.0  female        married          none      none   
14     15.0    male         single  professional      none   
15     16.0  female        married  professional      none   
16     17.0    male        married  professional        BA   
17     18.0  female        married  professional      none   
18     19.0  female         single  professional       PhD   
19     20.0  female        married  professional      none   
20     21.0  female        married          none        MA   
21     22.0    male         single  professional      none   
22     23.0  female        married          none      none   
23     24.0    male        married          none      none   
24     25.0    male         single  professional        BA   
25     26.0    male         single          none      none   
26     27.0    male         single          none      none   
27     28.0    male        married  professional      none   
28     29.0  female        married  professional      none   
29     30.0  female        married          none      none   
..      ...     ...            ...           ...       ...   
970   971.0  female        married          none        MA   
971   972.0    male        married  professional       PhD   
972   973.0    male        married          none        BA   
973   974.0    male        married          none      none   
974   975.0  female        married          none      none   
975   976.0  female        married          none        MA   
976   977.0  female        married          none       PhD   
977   978.0    male        married  professional      none   
978   979.0  female         single  professional        MA   
979   980.0    male         single          none      none   
980   981.0  female        married          none        BA   
981   982.0  female        married  professional      none   
982   983.0  female        married  professional      none   
983   984.0    male        married          none      none   
984   985.0    male        married          none      none   
985   986.0  female         single  professional      none   
986   987.0    male        married          none        BA   
987   988.0    male         single  professional        MA   
988   989.0    male        married          none        MA   
989   990.0    male         single  professional      none   
990   991.0    male        married  professional      none   
991   992.0  female        married          none        BA   
992   993.0  female         single  professional      none   
993   994.0    male        married  professional      none   
994   995.0    male        married  professional        BA   
995   996.0  female         single  professional      none   
996   997.0    male        married  professional        MA   
997   998.0    male        married  professional      none   
998   999.0    male        married          none        BA   
999  1000.0  female        married  professional      none   

     Annual Income (x1000 $)   Age        Location Purchasing Decision-maker  \
0                       49.0  30.0         Florida                    family   
1   

In [32]:
oldButGoodies = df.loc[(df["Age"] > 65) & (df["TV Viewing (hours/day)"] > 6)]

In [33]:
len(df)

1003

In [34]:
print("Annual Income for old people that watch a lot of TV")
print("number:", len(oldButGoodies))
print("range: ",oldButGoodies['Annual Income (x1000 $)'].min()," to ",oldButGoodies['Annual Income (x1000 $)'].max())
print("Average: ",oldButGoodies['Annual Income (x1000 $)'].mean())

Annual Income for old people that watch a lot of TV
number: 20
range:  41.0  to  55.0
Average:  48.6


In [35]:
df["Annual Income (x1000 $)"].corr(df["Age"], method="pearson")**2

0.019865189757789708

Only 14% of variation in annual income is explained by variation in age. youngins are getting loaded

In [36]:
df = pd.read_csv("TiVoData.csv")
dfGenderCoded = df
dfGenderCoded.head()
dfGenderCoded["Gender"] = dfGenderCoded['Gender'].map({'female': 1, 'male': 0})
df.head()


ID  Gender Marital Status   Work Status Education  \
0  1.0     0.0        married  professional      none   
1  2.0     0.0         single          none      none   
2  3.0     0.0        married  professional        BA   
3  4.0     0.0        married          none       PhD   
4  5.0     1.0         single          none      none   

   Annual Income (x1000 $)   Age       Location Purchasing Decision-maker  \
0                     49.0  30.0        Florida                    family   
1                     46.0  36.0        Alabama                    single   
2                     58.0  66.0  Massachusetts                    family   
3                     51.0  78.0       New York                    family   
4                     46.0  52.0        Montana                    single   

         Purchasing Location  Monthly Electronics Spend  \
0  mass-consumer electronics                       35.0   
1  mass-consumer electronics                       35.0   
2           specialty stores                       64.0   
3  mass-consumer electronics                       33.0   
4  mass-consumer electronics                       45.0   

   Monthly Household Spend  Purchasing Frequency (every x months)  \
0                    150.0                                   13.0   
1                    163.0                                   26.0   
2                    103.0                                   13.0   
3                    154.0                                   22.0   
4                    161.0                                   47.0   

  Technology Adoption  TV Viewing (hours/day)  \
0                late                     2.0   
1                late                    10.0   
2               early                     0.0   
3                late                     5.0   
4                late                     2.0   

                             Favorite feature  
0  saving favorite shows to watch as a family  
1  saving favorite shows to watch as a family  
2                               time shifting  
3  saving favorite shows to watch as a family  
4  saving favorite shows to watch as a family

In [38]:
dfGenderCoded["Annual Income (x1000 $)"].corr(dfGenderCoded["Gender"])**2

0.004552554812884768

6% of variation in Annual Income is attributable to age

*number used to code gender does not matter because we're looking at correlation info. Changing the number used would change the range of the factor we are looking at but it would not actually change the variation within the dataset

In [41]:
print(df["Purchasing Frequency (every x months)"].corr(df["Age"])**2, "of variation in # of months per purchase is explained by variation in age")

1.6708598227799212e-06 of variation in # of months per purchase is explained by variation in age


In [40]:
print(df["Annual Income (x1000 $)"].corr(df["TV Viewing (hours/day)"])**2, "of variation in annual income is explained by variation in tv viewing")

0.0071124339130305 of variation in annual income is explained by variation in tv viewing


In [43]:
df = pd.read_csv("TiVoData.csv")
dfEdCoded = df
dfEdCoded.head()
dfEdCoded["Education"] = dfEdCoded["Education"].map({'none': 0, 'BA': 1, 'MA':2,'PhD':3})
dfEdCoded["Favorite feature"] = dfEdCoded["Favorite feature"].map({'cool gadget': 0, 'time shifting': 1, 'schedule control':2,'saving favorite shows to watch as a family':3, 'programming/interactive features':4})
dfEdCoded.head()


ID  Gender Marital Status   Work Status  Education  \
0  1.0    male        married  professional        0.0   
1  2.0    male         single          none        0.0   
2  3.0    male        married  professional        1.0   
3  4.0    male        married          none        3.0   
4  5.0  female         single          none        0.0   

   Annual Income (x1000 $)   Age       Location Purchasing Decision-maker  \
0                     49.0  30.0        Florida                    family   
1                     46.0  36.0        Alabama                    single   
2                     58.0  66.0  Massachusetts                    family   
3                     51.0  78.0       New York                    family   
4                     46.0  52.0        Montana                    single   

         Purchasing Location  Monthly Electronics Spend  \
0  mass-consumer electronics                       35.0   
1  mass-consumer electronics                       35.0   
2           specialty stores                       64.0   
3  mass-consumer electronics                       33.0   
4  mass-consumer electronics                       45.0   

   Monthly Household Spend  Purchasing Frequency (every x months)  \
0                    150.0                                   13.0   
1                    163.0                                   26.0   
2                    103.0                                   13.0   
3                    154.0                                   22.0   
4                    161.0                                   47.0   

  Technology Adoption  TV Viewing (hours/day)  Favorite feature  
0                late                     2.0               3.0  
1                late                    10.0               3.0  
2               early                     0.0               1.0  
3                late                     5.0               3.0  
4                late                     2.0               3.0

In [44]:
print(dfEdCoded["Education"].corr(dfEdCoded["Favorite feature"])**2, "of variation in education is explained by variation in favorite feature")

0.13953354134201412 of variation in education is explained by variation in favorite feature


In [45]:
print(dfEdCoded["Monthly Electronics Spend"].corr(dfEdCoded["Monthly Household Spend"])**2, "of variation in spend  is explained by variation in house spend")

0.6461037171854618 of variation in spend  is explained by variation in house spend


In [5]:
dfE = pd.read_csv("TiVoEncoded.csv")

what's high enough correlation to ignore one? is .37 high enough?